pkg/bmc/idrac.go

```diff
@@ -0,0 +1,56 @@
+package bmc
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type iDracAccessDetails struct {
+	bmcType  string
+	portNum  string
+	hostname string
+	path     string
+}
+
+func (a *iDracAccessDetails) Type() string {
+	return a.bmcType
+}
+
+// NeedsMAC returns true when the host is going to need a separate
+// port created rather than having it discovered.
+func (a *iDracAccessDetails) NeedsMAC() bool {
+	return false
+}
+
+func (a *iDracAccessDetails) Driver() string {
+	return "idrac"
+}
+
+// DriverInfo returns a data structure to pass as the DriverInfo
+// parameter when creating a node in Ironic. The structure is
+// pre-populated with the access information, and the caller is
+// expected to add any other information that might be needed (such as
+// the kernel and ramdisk locations).
+func (a *iDracAccessDetails) DriverInfo(bmcCreds Credentials) map[string]interface{} {
+	host := a.hostname
+	if strings.ContainsRune(host, ':') {
+		// Hostname is an IPv6 address
+		host = fmt.Sprintf("[%s]", host)
```

Comment:
I'm surprised the URL class doesn't do this wrapping for us.

Reply:
Yeah, it's annoying. It seems to be more designed for going in the other direction (splitting a URL). There is the [Hostname()](https://golang.org/pkg/net/url/#URL.Hostname) function that extracts the IPv6 address, but no corresponding method for constructing the Host from a DNS/IPv4/IPv6 addresss + port :(